lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -1199,6 +1212,72 @@ TfToken MayaDescriptorToToken(const MVertexBufferDescriptor& descriptor)
 
 } // anonymous namespace
 
+class HdVP2Material::TextureLoadingTask
+{
+public:
+    TextureLoadingTask(
+        HdVP2Material*     parent,
+        HdSceneDelegate*   sceneDelegate,
+        const std::string& path,
+        const VtValue&     fallbackValue)
+        : _parent(parent)
+        , _sceneDelegate(sceneDelegate)
+        , _path(path)
+        , _fallbackValue(fallbackValue)
+    {
+    }
+
+    ~TextureLoadingTask() = default;
+
+    void Load()
+    {
+        if (_started.load()) {
```

Comment:
This check still has races in it. If two rprims share a material they could both enter this code in parallel. The call to _started.load() here and _started = true below will have a single total order consistent in all threads, multiple threads could load started before any thread sets the value true, leading to multiple idle tasks loading the same texture.

I think it would probably be okay, because the 2nd task would hit the cache, but it's still better to fix this.

I believe the correct approach would be something like:

```
if (_started.exchange(true)) {
    return;
}
```

That way if the value was already true we'll return right away, and exactly one thread will get a return value of false and create the idle task.

Reply:
Updated.